data/dbHelpers.js

```diff
@@ -0,0 +1,72 @@
+const knex = require('knex');
+
+const dbConfig = require('../knexfile');
+
+const db = knex(dbConfig.development);
+
+module.exports = {
+  getProjectById,
+  getProjects,
+  getActionsByProjectId,
+  addProject,
+  addAction,
+  deleteProject,
+  deleteAction
+}
+
+// Projects helper functions ------------------------
+// retrieve project by id number
+function getProjectById(id) {
+  return db('projects')
+    .where({
+      id: Number(id)
+    })
+    .first();
+}
+
+// retrieve all projects
+function getProjects() {
+  return db('projects')
+}
+
+// add new project
+function addProject(project) {
+  return db('projects')
+    .insert(project)
+    .then(ids => ({
+      id: ids[0]
```

Comment:
Another way of getting the first element out of an array is with [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Array_destructuring):
```javascript
.then( ([ id ]) => ({ id }) );
```